### ISSUE TITLE
More imgui widgets

### ISSUE THREAD

**Issue by dromer** [DISTRHO/DPF-Widgets#8]
Thought to collect some possible imgui widgets here:

https://github.com/cmdwtf/imgui_toggle - nice looking toggle switches - 0BSD
https://github.com/dalerank/imspinner - collection of animated spinner icons - MIT

**Comment by dromer** [comment]
Hmm, seems imgui_toggle breaks rules 1 and 6 since it contains more than one header and has several implementation files.

Unfortunate, since these do look quite nice.

**Comment by dromer** [comment]
Seems imspinner requires c++14 to work.

**Comment by falkTX** [comment]
This is not going to scale if we keep adding more widgets to imgui stuff. this toggle and knob are quite essential, so I let them pass, just dont assume it is always the case.

Regarding C++14 requirement, yeah it is preferred we keep it C++11 so the code needs tweaking.
Other imgui based things are C++11, so let's not bump the requirement just for 1 single line of code.

**Comment by dromer** [comment]
Yeah agree, besides those spinners are not critical in any way just thought they looked cool ;)

Actually the imgui_toggle says it might be included in upstream imgui at some point: https://github.com/cmdwtf/imgui_toggle#future-considerations - will probably take a while, but by then we can clean it up from the inclusion perhaps.

**Comment by falkTX** [comment]
Personally think the spinners are quite optional and we dont need them by default.

Out of the other widgets I have seen, only the nodes one would be ideal to have, but it needs some tweaking to work well for audio graph/patchbay purposes.

We can always manually include imgui based widgets on plugin gui code, alike done in master_me with the implot, see https://github.com/trummerschlunk/master_me/blob/master/plugin/widgets/Histogram.hpp

**Comment by dromer** [comment]
re node editor, which of these do you think is better?

- https://github.com/thedmd/imgui-node-editor
- https://github.com/Nelarius/imnodes

**Comment by dromer** [comment]
We were discussing sending midi notes from the UI and I came across: https://github.com/shric/midi/blob/master/src/Piano.cpp

![](https://raw.githubusercontent.com/shric/midi/master/doc/screenshots/midi-screenshot.jpg)
